pkg/storage/governance_request_feedback.go

```diff
@@ -0,0 +1,129 @@
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// CreateGovernanceRequestFeedback creates a governance request feedback record in the database
+func (s *Store) CreateGovernanceRequestFeedback(ctx context.Context, requestFeedback *models.GovernanceRequestFeedback) (*models.GovernanceRequestFeedback, error) {
+	// validate first
+	isFeedbackValid := validateSourceActionTargetFormCombination(*requestFeedback)
+	if !isFeedbackValid {
+		errorMessage := fmt.Sprintf(
+			"Invalid combination of source action and target feedback; %v is an invalid target form for a %v action",
+			requestFeedback.TargetForm,
+			requestFeedback.SourceAction,
+		)
+		appcontext.ZLogger(ctx).Error(errorMessage)
+		return nil, errors.New(errorMessage)
+	}
+
+	if requestFeedback.ID == uuid.Nil {
+		requestFeedback.ID = uuid.New()
+	}
+
+	const governanceRequestFeedbackCreateSQL = `
+		INSERT INTO governance_request_feedback(
+			id,
+			intake_id,
+			feedback,
+			source_action,
+			target_form,
+			created_by
+		) VALUES (
+			:id,
+			:intake_id,
+			:feedback,
+			:source_action,
+			:target_form,
+			:created_by
+		) RETURNING *;
+	`
+
+	stmt, err := s.db.PrepareNamed(governanceRequestFeedbackCreateSQL)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(
+			fmt.Sprintf("Failed to create governance request feedback with error %s", err),
+			zap.Error(err),
+			zap.String("user", requestFeedback.CreatedBy),
+		)
+		return nil, err
+	}
+
+	retFeedback := models.GovernanceRequestFeedback{}
+
+	err = stmt.Get(&retFeedback, requestFeedback)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(
+			fmt.Sprintf("Failed to create governance request feedback with error %s", err),
+			zap.Error(err),
+			zap.String("user", requestFeedback.CreatedBy),
+		)
+		return nil, err
+	}
+
+	return &retFeedback, nil
+}
+
+// GetGovernanceRequestFeedbacksByIntakeID returns all governance request feedback items for a given system intake
+func (s *Store) GetGovernanceRequestFeedbacksByIntakeID(ctx context.Context, intakeID uuid.UUID) ([]*models.GovernanceRequestFeedback, error) {
+	feedbacks := []*models.GovernanceRequestFeedback{}
+
+	stmt, err := s.db.PrepareNamed(`SELECT * FROM governance_request_feedback WHERE intake_id = :intake_id`)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(
+			"Failed to fetch governance request feedbacks",
+			zap.Error(err),
+			zap.String("intakeID", intakeID.String()),
+		)
+		return nil, err
+	}
+
+	arg := map[string]interface{}{"intake_id": intakeID}
+
+	err = stmt.Select(&feedbacks, arg)
+	if err != nil {
+		appcontext.ZLogger(ctx).Error(
+			"Failed to fetch governance request feedbacks",
+			zap.Error(err),
+			zap.String("intakeID", intakeID.String()),
+		)
+		return nil, &apperrors.QueryError{
+			Err:       err,
+			Model:     []*models.GovernanceRequestFeedback{},
+			Operation: apperrors.QueryFetch,
+		}
+	}
+
+	return feedbacks, nil
+}
+
+func validateSourceActionTargetFormCombination(feedback models.GovernanceRequestFeedback) bool {
```

Comment:
This function seems to work well, but I wonder if it would make more sense to live with the definition of the struct itself? Perhaps as a receiver function in the models package? I think of the storage package specifically being responsible for reading and writing to that database. What do you think?

Reply:
No longer applicable due to enforcing validation in SQL.